UI/EncyclopediaDetailPanel.cpp

```diff
@@ -260,7 +260,7 @@ namespace {
         }
         else if (dir_name == "ENC_BUILDING_TYPE") {
             for (const auto& [building_name, building_type] : GetBuildingTypeManager()) {
-                if (!exclude_custom_categories_from_dir_name || DetermineCustomCategory(building_type->Tags()).empty()) {
+                if (!boost::starts_with(building_name,"BLD_COL_") && (!exclude_custom_categories_from_dir_name || DetermineCustomCategory(building_type->Tags()).empty())) {
```

Comment:
this is adding some special case logic that depends on particular naming of content in scripts, so doesn't fit in this pull request and may not be acceptable at all (vs. specific-purpose tags or somesuch).

Reply:
understood.

Reply:
i have reverted that and made a tidy commit message. i promise to read up on tags and learn from older commits before trying something like that again.

Reply:
seems to me that at some other places in the code something similar is done, e.g. see screenshots attached.
![スクリーンショット 2022-01-08 18 00 49](https://user-images.githubusercontent.com/37298400/148652960-dc8b84ef-2f8b-44c1-b91b-c308e81a9c89.png)
![スクリーンショット 2022-01-08 17 58 00](https://user-images.githubusercontent.com/37298400/148652963-2e4445bd-243e-4609-9afa-b3e46acfbdc4.png)
and yes, i know, just because others did wrong does not allow me to do wrong, am just saying, is all .. am after all still quite noobish around here ;)

Reply:
that's python, not the core C++ engine... different rules or guidelines apply.